SimpleRss/SimpleRss/Services/RssService.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  RssService.swift
+//  SimpleRss
+//
+//  Created by Voldem on 12/12/18.
+//  Copyright © 2018 Vladimir Koptev. All rights reserved.
+//
+
+import Foundation
+
+class RssService: NSObject {
+    private struct TagConstants {
+        static let item = "item"
+        static let link = "url"
+        static let title = "title"
+        static let pubDate = "pubDate"
+        static let description = "description"
+        static let mediaDict = "media:content"
+        static let close = "/>"
+        static let brOpen = "<br"
+    }
+    
+    private struct ErrorConstants {
+        static let errorMessage = "Error while parsing"
+        static let noHandlerText = "No handler for "
+    }
+    
+    var feedList = [Feed]()
+    var feed: Feed?
+    var parser: RssParser?
+    
+    func getFeed(for link: String?, withCallback completionHandler: @escaping(_ result: [Feed]?, _ error: Error?) -> Void) {
+        guard let link = link, let url = URL(string: link) else { return }
+        parser = RssParser()
+        parser?.parse(url) { [weak self] (result, error) in
+            guard let strongSelf = self else { completionHandler(nil, nil)
```

Comment:
Calling completionHandler in this case is not needed (unless you can provide an example of a case where completionHandler needs to be called after`RssService` is deleted)